File: erp/src/main/java/com/erp/model/PurchaseOrder.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "purchase_order")
@Table(name = "purchase_order")
public class PurchaseOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "purchase_order_code")
    private Long purchaseOrderCode;

    @Column(name = "supplier_code")
    private Long supplierCode;

    @Column(name = "supplier_name")
    private String supplierName;

    @Column(name = "order_date")
    @Temporal(TemporalType.DATE)
    private Date orderDate;

    @Column(name = "received_date")
    @Temporal(TemporalType.DATE)
    private Date receivedDate;

    @Column(name = "product_code")
    private Long productCode;

    @Column(name = "product_name")
    private String productName;
    
    @Column(name = "unit_price")
    private Double unitPrice;    

    @Column(name = "purchase_quantity")
    private Integer purchaseQuantity;

    @Column(name = "total_amount")
    private Double totalAmount;

    @Column(name = "payment_status")
    private String paymentStatus;

    @Column(name = "status")
    private String status;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getPurchaseOrderCode() {
		return purchaseOrderCode;
	}

	public void setPurchaseOrderCode(Long purchaseOrderCode) {
		this.purchaseOrderCode = purchaseOrderCode;
	}

	public Long getSupplierCode() {
		return supplierCode;
	}

	public void setSupplierCode(Long supplierCode) {
		this.supplierCode = supplierCode;
	}

	public String getSupplierName() {
		return supplierName;
	}

	public void setSupplierName(String supplierName) {
		this.supplierName = supplierName;
	}

	public Date getOrderDate() {
		return orderDate;
	}

	public void setOrderDate(Date orderDate) {
		this.orderDate = orderDate;
	}

	public Date getReceivedDate() {
		return receivedDate;
	}

	public void setReceivedDate(Date receivedDate) {
		this.receivedDate = receivedDate;
	}

	public Long getProductCode() {
		return productCode;
	}

	public void setProductCode(Long productCode) {
		this.productCode = productCode;
	}

	public String getProductName() {
		return productName;
	}

	public void setProductName(String productName) {
		this.productName = productName;
	}

	public Double getUnitPrice() {
		return unitPrice;
	}

	public void setUnitPrice(Double unitPrice) {
		this.unitPrice = unitPrice;
	}

	public Integer getPurchaseQuantity() {
		return purchaseQuantity;
	}

	public void setPurchaseQuantity(Integer purchaseQuantity) {
		this.purchaseQuantity = purchaseQuantity;
	}

	public Double getTotalAmount() {
		return totalAmount;
	}

	public void setTotalAmount(Double totalAmount) {
		this.totalAmount = totalAmount;
	}

	public String getPaymentStatus() {
		return paymentStatus;
	}

	public void setPaymentStatus(String paymentStatus) {
		this.paymentStatus = paymentStatus;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

}

File: ebs/lib/model/PurchaseOrder.dart
class PurchaseOrder {
  final int? id;
  final int? purchaseOrderCode;
  final int? supplierCode;
  final String? supplierName;
  final DateTime? orderDate;
  final DateTime? receivedDate;
  final int? productCode;
  final String? productName;
  final double? unitPrice;
  final int? purchaseQuantity;
  final double? totalAmount;
  final String? paymentStatus;
  final String? status;

  PurchaseOrder({
    this.id,
    this.purchaseOrderCode,
    this.supplierCode,
    this.supplierName,
    this.orderDate,
    this.receivedDate,
    this.productCode,
    this.productName,
    this.unitPrice,
    this.purchaseQuantity,
    this.totalAmount,
    this.paymentStatus,
    this.status,
  });

  factory PurchaseOrder.fromJson(Map<String, dynamic> json) {
    return PurchaseOrder(
      id: json['id'],
      purchaseOrderCode: json['purchaseOrderCode'],
      supplierCode: json['supplierCode'],
      supplierName: json['supplierName'],
      orderDate: json['orderDate'] != null ? DateTime.parse(json['orderDate']) : null,
      receivedDate: json['receivedDate'] != null ? DateTime.parse(json['receivedDate']) : null,
      productCode: json['productCode'],
      productName: json['productName'],
      unitPrice: json['unitPrice'] != null ? (json['unitPrice'] as num).toDouble() : null,
      purchaseQuantity: json['purchaseQuantity'],
      totalAmount: json['totalAmount'] != null ? (json['totalAmount'] as num).toDouble() : null,
      paymentStatus: json['paymentStatus'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'purchaseOrderCode': purchaseOrderCode,
      'supplierCode': supplierCode,
      'supplierName': supplierName,
      'orderDate': orderDate?.toIso8601String(),
      'receivedDate': receivedDate?.toIso8601String(),
      'productCode': productCode,
      'productName': productName,
      'unitPrice': unitPrice,
      'purchaseQuantity': purchaseQuantity,
      'totalAmount': totalAmount,
      'paymentStatus': paymentStatus,
      'status': status,
    };
  }
}